src/libratbag.h

```diff
@@ -1320,13 +1343,169 @@ ratbag_button_get_special(struct ratbag_button *button);
+ */
+enum ratbag_led_mode {
+	RATBAG_LED_OFF = 0,
+	RATBAG_LED_ON,
+	RATBAG_LED_CYCLE,
+	RATBAG_LED_BREATHING,
+};
+/**
+ * @ingroup led
+ *
  * LED types, usually based on their physical location
  */
 enum ratbag_led_type {
 	RATBAG_LED_TYPE_UNKNOWN = -1,
 	RATBAG_LED_TYPE_LOGO = 0,
 	RATBAG_LED_TYPE_SIDE
 };
+
+/**
+ * @ingroup led
+ *
+ * This function returns the type for ratbag_led.
+ *
+ * @param led A previously initialized ratbag LED
+ * @return The LED type @ref enum ratbag_led_type
+ *
+ * @see ratbag_led_set_mode
+ */
+enum ratbag_led_type
+ratbag_led_get_type(struct ratbag_led *led);
+/**
+ * @ingroup led
+ *
+ * This function returns the mode for ratbag_led.
+ *
+ * @param led A previously initialized ratbag LED
+ * @return The LED mod @ref enum ratbag_led_mode
+ *
+ * @see ratbag_led_set_mode
+ */
+enum ratbag_led_mode
+ratbag_led_get_mode(struct ratbag_led *led);
+/**
+ * @ingroup led
+ *
+ * This function returns the led color.
+ *
+ * @param led A previously initialized ratbag LED
+ * @return The LED color in @ref enum ratbag_led_mode
+ *
+ * @see ratbag_led_set_color
+ */
+struct ratbag_color
+ratbag_led_get_color(struct ratbag_led *led);
+/**
+ * @ingroup led
+ *
+ * This function returns the LED effect rate.
+ *
+ * @param led A previously initialized ratbag LED
+ * @return The LED rate in Hz, can be 100 - 20000
+ *
+ * @see ratbag_led_set_effect_rate
+ */
+int
+ratbag_led_get_effect_rate(struct ratbag_led *led);
+/**
+ * @ingroup led
+ *
+ * This function returns the LED brightness.
+ *
+ * @param led A previously initialized ratbag LED
+ * @return The LED brightness 0 - 255
+ *
+ * @see ratbag_led_get_brightness
+ */
+unsigned int
+ratbag_led_get_brightness(struct ratbag_led *led);
+/**
+ * @ingroup led
+ *
+ * Return a reference to the LED given by the index. The order of the
+ * LEDs is device-specific though.
+ *
+ * The LED is refcounted with an initial value of at least 1.
+ * Use ratbag_led_unref() to release the LED.
+ *
+ * @param profile A previously initialized ratbag profile
+ * @param index The index of the LED
+ *
+ * @return A LED context, or NULL if the LED does not exist.
+ *
+ * @see ratbag_device_get_profile
+ */
+struct ratbag_led *
+ratbag_profile_get_led(struct ratbag_profile *profile, unsigned int index);
+
+/**
+ * @ingroup led
+ *
+ * this function sets the LED mode.
+ *
+ * @param led A previously initialized ratbag LED
+ * @param mode LED mode @ref enum ratbag_led_mode.
+ * @return 0 on success or an error code otherwise.
+ *
+ * @see ratbag_led_get_mode
+ */
+enum ratbag_error_code
+ratbag_led_set_mode(struct ratbag_led *led, enum ratbag_led_mode mode);
+
+/**
+ * @ingroup led
+ *
+ * If the LED's mode is @ref RATBAG_LED_ON or @ref RATBAG_LED_BREATHING
+ * then this function sets the LED color, otherwise it has no effect.
+ *
+ * @param led A previously initialized ratbag LED
+ * @param color A LED color in hex format (ffffff).
```

Comment:
whether it's in hex or not shouldn't matter here, right. this looks like a leftover